src/internal/partial_solution.rs

```diff
@@ -501,7 +501,7 @@ impl<P: Package, VS: VersionSet> PackageAssignments<P, VS> {
         let idx = self
             .dated_derivations
             .as_slice()
-            .partition_point(|dd| dd.accumulated_intersection.intersection(start_term) != empty);
+            .partition_point(|dd| !dd.accumulated_intersection.is_disjoint(start_term));
```

Comment:
The other place where we could use these new methods is in `relation` in that context we want to know `is_disjoint` and `subset_of` for the same operands. The question becomes can the combination be computed faster if we asked them at the same time? If so does it justify having all three methods? @konstin, you spent the most time looking at optimizing the methods recently, what are your thoughts?

Reply:
I don't see `relation_with` in the traces and `relation` seems to be almost exclusively map lookups:

![image](https://github.com/pubgrub-rs/pubgrub/assets/6826232/ed2313a0-e894-4f23-a18c-108b4dd810f8)

Reply:
If map lookups are a significant bottleneck I would love to see what performance you getting with:
https://github.com/pubgrub-rs/pubgrub/tree/p_id

Reply:
It's not package ids, it's incompatibility ids in `self.contradicted_incompatibilities.contains_key(&incompat_id)` (5.8% in this run)

![image](https://github.com/pubgrub-rs/pubgrub/assets/6826232/790bf849-55f4-470a-88d8-e4fceeacf063)

Reply:
Yes, that is an expensive call, but it is not the hashmap lookup that is slowing `relation`.
`terms_intersection_for_package` is a lookup by package ID.

Reply:
I tried to cherry pick the commit but it unfortunately has to main merge conflicts with our main for me to try

Reply:
Okay, we will experiment with rebasing that branch once we have this merged. Sorry for getting distracted.